dev_tools/scripts/format_header.sh

```diff
@@ -0,0 +1,253 @@
+#!/bin/bash
+
+convert_header () {
+    local regex="^(#include )(<|\")(.*)(\"|>)$"
+    if [[ $@ =~ ${regex} ]]; then
+        header_file="${BASH_REMATCH[3]}"
+        if [ -f "${header_file}" ]; then
+            if [[ "${header_file}" =~ ^ginkgo ]]; then
+                echo "#include <${header_file}>"
+            else
+                echo "#include \"${header_file}\""
+            fi
+        elif [ "${header_file}" = "matrices/config.hpp" ]; then 
+            echo "#include \"${header_file}\""
+        else
+            echo "#include <${header_file}>"
+        fi
+    else
+        echo "$@"
+    fi
+}
+
+get_header_def () {
+    local regex="\.(hpp|cuh)"
+    if [[ $@ =~ $regex ]]; then
+        local def=$(echo "$@" | sed -E "s~include/ginkgo/~~g;s~/|\.~_~g")
+        def=$(echo GKO_${def^^}_)
+        echo $def
+    else
+        echo ""
+    fi
+}
+
+add_regroup () {
+    cp .clang-format .clang-format.temp
+    sed -i "s~\.\.\.~~g" .clang-format
+    cat dev_tools/scripts/regroup >> .clang-format
+    echo "..." >> .clang-format
+}
+
+remove_regroup () {
+    mv .clang-format.temp .clang-format
+}
+
+get_include_regex () {
+    local file="$1"
+    local core_suffix=""
+    local path_prefix=""
+    local path_ignore="0"
+    local fix_include=""
+    local remove_test="false"
+    local item_regex="^-\ +\"(.*)\""
+    local path_prefix_regex="PathPrefix:\ +\"(.*)\""
```

Comment:
Can you maybe add a comment explaining the effect of these configuration options? Some are obvious, for others you might have to look closer into the code.

Reply:
Do you mean this comment?
I write some comments to explain the configuration option [here](https://github.com/ginkgo-project/ginkgo/pull/447/files#diff-e7e36e51d1a12bf3793383440bca985cR45-R61)
Or, do I misunderstand what you mean?

Reply:
Thanks, that was what I was looking for!